### DISCUSSION TITLE
Accepting any input on an custom node input?

### DISCUSSION THREAD

**Discussion by darkpixel** [comfyanonymous/ComfyUI#4830] [Q&A] [answered]
I've dug around the code and can't really find an authoritative source for input types.
You can figure out from examples and other custom modules about accepted input types for things like STRING, INT, FLOAT, BOOL, etc...

I had a custom node that had an input type of ```"*"```.  I would look at whatever the input was (string, float, array, etc...) and do stuff with it.  I would output a STRING for use in the workflow.

It's worked fine for the last year.  But a recent change (some time in the last 30-90 days) now causes the console to throw:
```
 - Return type mismatch between linked nodes: evaluate_input <whatever data was input>  != *
```

I've tried setting the input to ```""``` and ```"ANY"``` and ```"ALL"```.  Nothing works.

How can I tell ComfyUI that I want to accept anything under the sun as input?

Also...I think it would be good to have a list of common standard input types that are generally accepted.

**Comment by ltdrdata** [comment, marked as answer]
"*" was never actually allowed. If it worked, it was likely in a blind spot of the validation.
To bypass this, custom node developers are using the `any_typ` trick.

https://github.com/ltdrdata/ComfyUI-Inspire-Pack/blob/25ae8d522f781f694bf736f87945f2291da531b4/inspire/libs/utils.py#L214

https://github.com/ltdrdata/ComfyUI-Inspire-Pack/blob/25ae8d522f781f694bf736f87945f2291da531b4/inspire/backend_support.py#L110

https://github.com/ltdrdata/ComfyUI-Inspire-Pack/blob/25ae8d522f781f694bf736f87945f2291da531b4/inspire/prompt_support.py#L716

**Reply by SayanoAI** [reply]
Does this work with types other than "*" (e.g. "INT,FLOAT" for any numeric input)? Also, does it work with the native get/set nodes?

**Reply by ltdrdata** [reply]
A wildcard input allows any type of connection. 

In the case of primitive nodes, connections are not allowed if the input type is not a primitive type, as this is restricted by the primitive node itself.

In my node pack, once a connection is made, I employ a method that forcibly assigns a type. After the type is assigned, the structure becomes one where connections are possible.

**Comment by darkpixel** [comment]
Thanks for the pointer @ltdrdata.  I was able to get things working!
